lib/tasks/update_search_trigrams.rake

```diff
@@ -1,31 +1,54 @@
 namespace :search do
 
+  def rkey(resource)
+    "rake-task-update-#{resource}-tsvectors"
+  end
+
+  def start_id(resource)
+    start_batch = if redis.exists(rkey(resource))
```

Comment:
I think you might have to pass redis in here or encapsulate this logic to a class that has redis
```
class RedisCheckpointer
  attr_reader :redis, :start

  def initialize
    # make sure the URL doesn't have the db number in it
    # or the explicit setting here overrides the url suffix
    @redis = Redis.new(host: ENV['REDIS_URL'], db: 15)
  end

  def start_id(resource)
....
  end

  def rkey(resource)
.....
  end
end
```

Reply:
ok :D